pinterest/teletraan
# Always prioritize deploy that does not need to wait

Status: waiting for r+

Problem:
---
Since, deploy are interliving, deployd ABORTs inprogress deploy by killing script when higher priority deploy kicks in. This is problematic since currently deploy currently could loose progress made.

Fix:
---

Priority order:
1. Consider deploy in STOP state with highest priority
2. Next, consider deploy which does not need to wait
3. Next, consider deploy based on priority order

Test:
---
Service A priority X and service B priority Y (Y > X)
1. Deploy service A followed by service B and, verify A runs through completion before B starts.
2. Deploy service A followed by service B and, verify B runs when A is stuck on host.
3. Deploy service A and service B and, verify B is deployed first then A.
4. Verify deploy order on new host
